chains/solana/contracts/programs/mcm/src/instructions/set_config.rs

```diff
@@ -110,6 +111,27 @@ pub fn set_config(
     config.group_quorums = group_quorums;
     config.group_parents = group_parents;
 
+    // clear_root is equivalent to overriding with a completely empty root
+    if clear_root {
+        let expiring_root = &mut ctx.accounts.expiring_root_and_op_count;
+        let root_metadata = &mut ctx.accounts.root_metadata;
+
+        // preserve the current op_count
+        let current_op_count = expiring_root.op_count;
+
+        // clear the expiring root while preserving op_count
+        expiring_root.root = [0u8; 32]; // clear root (equivalent to bytes32(0) in Solidity)
+        expiring_root.valid_until = 0; // clear timestamp
+        expiring_root.op_count = current_op_count;
```

Comment:
I see here that you're overwriting the entire account inner struct so as to fully clear it. A nice way of having the compiler also check for any missing attributes (e.g. in case we add more properties to the account in a future PR) is to use the account's `set_inner` method. You should be able to do something like the following:

```rs
expiring_root.set_inner(ExpiringRootAndOpCount {
  // set all props here. The compiler will check that you've not missed any property.
})
```

There are some places in the code where we're already doing this, if you want to look for other examples.

Reply:
> compiler also check for any missing attributes

Sounds great, I'll have that update in the following review-modification PR

Reply:
Updated in [`49c96d3` (#393)](https://github.com/smartcontractkit/chainlink-ccip/pull/393/commits/49c96d3423f6358a9bfd74f25161ef00326e2eaa)